src/User/UpdateProfile.jsx

```diff
@@ -1,17 +1,92 @@
-import React from "react";
+import React, { useState } from "react";
+import Input from "../Input";
+import { useHistory } from "react-router-dom";
 import {
-  StyledForm,
-  StyledCard
+	StyledForm,
+	StyledCard,
+	StyledCardHeading
 } from "../StyledFormComponents";
+import { updateUserById } from "../api";
 
-const UpdateProfile = () => {
-  return (
-      <StyledForm>
-          <StyledCard>
-              Edit Profile
-        </StyledCard>
-    </StyledForm>
-  );
+const UpdateProfile = ({
+	setUser,
+	id,
+	isLoading,
+	error,
+    sessionToken,
+    user
+}) => {
+	const [updatedUser, setUpdatedUser] = useState({});
+	const history = useHistory();
+	const resetUser = () => {
+        history.push(`/users/${id}`);
+    };
+    
+	if (isLoading) {
+		return <p>Loading...</p>;
+	}
+
+	if (error) {
+		return <p>An error has occurred...</p>;
+    }
+
+	const handleChange = event => {
+		const { name, value } = event.target;
+		setUpdatedUser({ ...updatedUser, [name]: value });
+	};
+
+	const submitForm = async () => {
+		if (Object.keys(updatedUser).length > 0)
```

Comment:
I think because I needed to find the length of the properties in the object? @tonymaynard did explain it to me whilst we were doing it but I can't quite recall...

Reply:
It's correct for what you're doing but I'd recommend going back over it with him / your helper / someone / googlefriend

Reply:
@karenzamani Yeah so we used it to find if there were any keys in the object, if `updatedUser = {}` there's no point in making an api call to with no data to patch!